### PR TITLE
Change Cholesky to be a struct like QR, LU, and LQ. Add condition num…

### PR DESCRIPTION
**Pull request by btracey**
…ber check to Solve routines

### CODE REVIEW

**Comment by btracey** [comment]
It's so nice to see greens the first time.

File: mat64/cholesky.go [hunk 1]
```diff
@@ -13,50 +15,65 @@ import (
...
+}
+
+// updateCond updates the condition number of the Cholesky decomposition. If
+// norm > 0, then that norm is used as the norm of the original matrix A, otherwise
+// the norm is estimated from the decompositon.
+func (c *Cholesky) updateCond(norm float64) {
+	n := c.chol.mat.N
+	work := make([]float64, 3*n)
+	if norm < 0 {
+		// This is an approximation. By the defintion of a norm, ||AB|| <= ||A|| ||B||.
```
Comment (kortschak) [inline mat64/cholesky.go]: definition

File: mat64/cholesky.go [hunk 2]
```diff
@@ -13,50 +15,65 @@ import (
...
-			Stride: n,
-			Diag:   blas.NonUnit,
-			Data:   use(t.mat.Data, n*n),
-		}
-		if upper {
-			t.mat.Uplo = blas.Upper
-		} else {
-			t.mat.Uplo = blas.Lower
-		}
+	if c.chol == nil {
```
Comment (kortschak) [inline mat64/cholesky.go]: Can't this just use the else case operation and lose the condition test?

**Comment by kortschak** [comment]
LGTM

**Pull request merged**